### PR TITLE
F#20 polish character controls

### PR DESCRIPTION
**Pull request by massieb**

### CODE REVIEW

**Comment by massieb** [comment]
PR requested by Jin to play around with the new mechanics. 

Hidden, can you please be co kind to give your thoughts on the new implementation and how i can improve that even further. I'm having a hard time to make this generic.

File: Assets/Scripts/Player/PlayerController2D.cs [hunk 1]
```diff
@@ -0,0 +1,202 @@
+using System;
+using System.Collections;
+using UnityEngine;
+
+public class PlayerController2D : MonoBehaviour
```
Comment (Dejin1) [inline Assets/Scripts/Player/PlayerController2D.cs]: Does this script overwrite the other PlayerController2D or other scripts you made will it break other scenes with the old control scripts?

File: Assets/Scripts/Player/PlayerController2D.cs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    {
+        animator = GetComponent<PlayerAnimator>();
+        state = new PlayerState();
+        rigidBody2D = GetComponent<Rigidbody2D>();
+    }
+
+    public void ProcessPlayerInput(PlayerInput playerInput)
+    {
+        this.input = playerInput;
+
```
Comment (Dejin1) [inline Assets/Scripts/Player/PlayerController2D.cs]: Extra un-needed spaces?

File: Assets/Scripts/Player/PlayerController2D.cs (hunk 1 above)
Comment (Dejin1) [reply Assets/Scripts/Player/PlayerController2D.cs]: I recently checked you used different scripts entirely which is great though the naming structure and not having them in a different folder than the current scripts is a bit confusing I would attempt to put your new player movement scripts in a folder for now till we transition to them. What do you think?

Comment (massieb) [reply Assets/Scripts/Player/PlayerController2D.cs]: Exactly, I made zero changes to all existing files. So this should give no problems with merging

File: Assets/Scripts/Player/PlayerController2D.cs (hunk 2 above)
Comment (massieb) [reply Assets/Scripts/Player/PlayerController2D.cs]: No it's intended to separate a setter with a method call

**Pull request merged**